arroba/tests/mst_test_suite.py

```diff
@@ -81,3 +101,30 @@ def test_diffs(self):
             self.assertEqual(created_list, testcase["results"]["created_nodes"], f"{testname} created_nodes") # currently fails!
             self.assertEqual(deleted_list, testcase["results"]["deleted_nodes"], f"{testname} deleted_nodes")
             # TODO: implement checks for proof_nodes, firehose_cids (test data hasn't been generated yet)
+
+    def test_diffs_inverse(self):
+        # we re-use the diff test cases but "backwards" - applying the op list
+        # to the initial MST see if we end up at the correct final MST
+        for testname, testcase in tqdm(self.diff_testcases.items()):
+            storage = MemoryStorage()
+            root_a = self.populate_storage_from_car(storage, testcase["inputs"]["mst_a"])
+            mst = MST.load(storage=storage, cid=root_a)
+
+            for op in testcase["results"]["record_ops"]:
+                if op["old_value"] and op["new_value"]: # update
+                    mst = mst.update(op["rpath"], CID.decode(op["new_value"]))
+                elif op["old_value"]: # delete
+                    mst = mst.delete(op["rpath"])
+                else: # create
+                    mst = mst.add(op["rpath"], CID.decode(op["new_value"]))
+
+            diff = null_diff(mst) # should get us a map of the complete new mst
+            root_b = mst.get_pointer()
+
+            with open(self.test_suite_base  + testcase["inputs"]["mst_b"], "rb") as car_b:
+                reference_root, reference_blocks = self.parse_car(car_b)
+
+            reference_cid_set = set(x[0] for x in reference_blocks) # just look at the cids from the car
+
+            self.assertEqual(root_b, reference_root, f"{testname} inverse: new root") # fails occasionally
+            self.assertEqual(diff.new_cids, reference_cid_set, f"{testname} inverse: new cid set") # basically always fails, I think I'm doing something wrong
```

Comment:
the gist of this test case is: take `mst_a`, apply the list of ops to it, and compare the result to `mst_b`. The list of CIDs I'm getting from `null_diff(mst).new_cids` looks *completely* different to what I expect, so I suspect I'm doing something wrong here (or maybe null_diff() is just very broken heh)